### PR TITLE
[Feature] [flink] Remove flink DataSet api (#1246)

### PR DESCRIPTION
**Pull request by suxinshuo**
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by xleoken** [comment]
Overall LGTM, cc @asdf2014

File: seatunnel-connectors/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/sink/DruidSink.java [hunk 1]
```diff
@@ -45,14 +47,16 @@
...
+    public DataStreamSink<Row> outputStream(FlinkEnvironment env, DataStream<Row> dataStream) {
+        DataStreamSink<Row> rowDataStreamSink = dataStream.addSink(new DruidSinkFunction<>(
+                new DruidOutputFormat<>(coordinatorURL, datasource, timestampColumn, timestampFormat, timestampMissingValue)));
         if (config.hasPath(PARALLELISM)) {
             int parallelism = config.getInt(PARALLELISM);
-            return dataSink.setParallelism(parallelism);
+            rowDataStreamSink.setParallelism(parallelism);
         }
-        return dataSink;
+        return null;
```
Comment (asdf2014) [inline seatunnel-connectors/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/sink/DruidSink.java]: May I ask why we don't return the sink instance here

Comment (xleoken) [reply seatunnel-connectors/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/sink/DruidSink.java]: In the main logic, we don't use the returned sink instance.

Comment (asdf2014) [reply seatunnel-connectors/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/sink/DruidSink.java]: Indeed, it would be better we improve the interface directly

Comment (xleoken) [reply seatunnel-connectors/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/sink/DruidSink.java]: > 

You are right.

Comment (suxinshuo) [reply seatunnel-connectors/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/sink/DruidSink.java]: I can change this function to void and then submit this pr again.

Comment (suxinshuo) [reply seatunnel-connectors/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/sink/DruidSink.java]: I'm done. @leo65535 @asdf2014

Comment (asdf2014) [reply seatunnel-connectors/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/sink/DruidSink.java]: @xsaffable Thanks for your contribution, let's wait the CI pass. Sorry, I don't have permission to help you to start the pending workflows. Maybe @leo65535 can help you out I think.

Comment (xleoken) [reply seatunnel-connectors/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/sink/DruidSink.java]: Thanks for your review @asdf2014

Comment (asdf2014) [reply seatunnel-connectors/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/sink/DruidSink.java]: @leo65535 You are welcome

Comment (suxinshuo) [reply seatunnel-connectors/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/sink/DruidSink.java]: Thank you @leo65535 @asdf2014 . But I see it seems to time out, can you help me

**Comment by suxinshuo** [comment]
@leo65535 Hi PTAL.Thx.

**Comment by wuchunfu** [comment]
@xsaffable Please resolve the conflicting files, thanks

**Comment by suxinshuo** [comment]
> @xsaffable Please resolve the conflicting files, thanks

Hello, i have resolved the conflict file. @wuchunfu

File: seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/stream/FlinkStreamSink.java [hunk 2]
```diff
@@ -21,17 +21,13 @@
...
-import javax.annotation.Nullable;
-
 /**
  * a FlinkStreamSink plugin will write data to other system using Flink DataStream API.
  */
 public interface FlinkStreamSink extends BaseFlinkSink {
 
-    @Nullable
-    DataStreamSink<Row> outputStream(FlinkEnvironment env, DataStream<Row> dataStream);
+    void outputStream(FlinkEnvironment env, DataStream<Row> dataStream);
```
Comment (1996fanrui) [inline seatunnel-apis/seatunnel-api-flink/src/main/java/org/apache/seatunnel/flink/stream/FlinkStreamSink.java:31]: Could we remove the FlinkStreamSink and  move the method to BaseFlinkSink? BaseFlinkSource is similar.

Because we don't distinguish the batch or stream, we can just keep the BaseFlinkSink. All flink sinks implement the BaseFlinkSink. How do you think?

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/pom.xml [hunk 3]
```diff
@@ -36,10 +36,25 @@
...
+            <artifactId>flink-table-planner_${scala.binary.version}</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-streaming-java_${scala.binary.version}</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-table-common</artifactId>
+            <version>${flink.version}</version>
```
Comment (1996fanrui) [inline seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/pom.xml:55]: Could the `${flink.version}` be removed?

File: seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/sink/DruidOutputFormat.java [hunk 4]
```diff
@@ -56,7 +56,7 @@
 import java.util.Arrays;
 import java.util.Collections;
 
-public class DruidOutputFormat extends RichOutputFormat<Row> {
+public class DruidOutputFormat<T> extends RichOutputFormat<T> {
```
Comment (1996fanrui) [inline seatunnel-connectors/seatunnel-connectors-flink/seatunnel-connector-flink-druid/src/main/java/org/apache/seatunnel/flink/sink/DruidOutputFormat.java:59]: From the writeRecord, we can see it just support Row, why add the `<T>` for DruidOutputFormat? Could you revert this change? If no, please add some details, thanks. 

I think this change will delay the Error until writeRecord. And it just log.ERROR, flink job will run normally, right?

File: seatunnel-transforms/seatunnel-transforms-flink/seatunnel-transform-flink-split/src/main/java/org/apache/seatunnel/flink/transform/Split.java [hunk 5]
```diff
@@ -20,21 +20,19 @@
...
 import org.apache.flink.api.scala.typeutils.Types;
 import org.apache.flink.streaming.api.datastream.DataStream;
 import org.apache.flink.types.Row;
 
 import java.util.List;
 
-public class Split implements FlinkStreamTransform, FlinkBatchTransform {
+public class Split implements FlinkStreamTransform {
 
     private Config config;
```
Comment (1996fanrui) [inline seatunnel-transforms/seatunnel-transforms-flink/seatunnel-transform-flink-split/src/main/java/org/apache/seatunnel/flink/transform/Split.java:37]: About the Transform refactor:

- Could you remove the FlinkStreamTransform? 
- rename the method name from processStream to process
- move the process to BaseFlinkTransform
- Change the BaseFlinkTransform from interface to abstract class
- move the `private Config config` and setter/getter to BaseFlinkTransform

After finish these changes, it can greatly reduce the duplication of code and remove the useless class.

**Review by 1996fanrui — COMMENTED** [review]
Hi @xsaffable , I have left some comments, please address these comments in your free time. 

Actually, I'm a new user of Seatunnel, I might miss something. So you can correct me if I'm wrong, thanks a lot.

**Comment by CalvinKirs** [comment]
no updates, if you have any updates, you can leave a message, and I will reopen it.
